enterprise/web.jsf.editor/src/org/netbeans/modules/web/jsf/editor/index/CompositeComponentModel.java

```diff
@@ -270,7 +270,7 @@ private static FileObject getResourcesDirectory(FileObject file) {
             if (folder.getName().equalsIgnoreCase("resources")) { //NOI18N
                 //check if its parent is META-INF
                 FileObject parent = folder.getParent();
-                if (parent != null && parent.getNameExt().startsWith("META-INF")) { //NOI18N
+                if (parent != null && parent.getName().startsWith("META-INF")) { //NOI18N
```

Comment:
This should not be necessary. FileObject#getName would be better called "getNameWithoutExtension" and FileObject#getNameExt returns the normal filename. Could you please double check?

Reply:
The inclusion of the "extension" is irrelevant here I guess, as you're only looking for `META-INF` at the beginning. That's probably why the original patch, as suggested on stackoverflow, changed from `getNameExt()` to `getName()`. Additional benefit is that you also get rid of a string concat by not using `getNameExt()`.

As for `FileObject` itself: do you really want to change such a "core" class? This would trigger a lot of changes all over the place, no?

Reply:
No I don't want to change FileObject, that sentence was meant as an explanation. I think FileObject is a broken abstraction, but indeed it so far into the base of netbeans, that it won't be changed. I prefer changes to be minimal. Not needing this change, dropping it is IMHO a good idea.

Reply:
`FileObject` indeed looks a bit, how to put it, hairy.